src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -119,6 +120,7 @@ public function send(string $body, array $headers, int $delay = 0): string
                 'available_at' => '?',
             ]);
 
+        $useDeprecatedConstants = !class_exists(Types::class);
```

Comment:
wouldn't it be better to use a static property initialized once as this triggers the class loader for every send and get (even twice)

Reply:
`self::$useDeprecatedConstants = self::$useDeprecatedConstants ?? class_exists(Types::class);` in the constructor (same in the other PR then)